### ISSUE TITLE
Better handling of cookbooks and recursion.

### ISSUE THREAD

**Issue by docwhat** [acrmp/foodcritic#250]
When developing cookbooks in a one-cookbook-per-repository way, foodcritic is really annoying:
1. It scans directories like `spec` as if they are cookbooks.
2. It doesn't understand that there is only one cookbook to check.
3. It scans `_spec.rb` files as they are recipes and complains about weird things.
4. You can't globally turn off checks

Without a project configuration, it means that foodcritic runs differently depending on how it is run:
- Vim (via Syntastic) produces errors about `spec/README.md` and bogus errors and `_spec.rb` files.
- `foodcritic` run at the root via shell produces errors about `spec/README.md'.
- `foodcritic` run in a subdirectory doesn't understand what's going on at all.
- Running via Guard produces different output, depending on how you customized your `Guardfile`.
- The `rake` task automatically ignores `specs/` and `features/`

I would suggest that `foodcritic` be changed to work one particular way:
1. If the current directory (or specified directory) contains `metadata.rb` or `metadata.json` then add it to a list of cookbooks and `GOTO 3`.
2. Search recursively for directories with `metadata.rb` or `metadata.json` and add them to a list of cookbooks. Do not descend below those directory with `metadata.{rb,json}` files.
3. If the cookbook list is empty, emit an error "No cookbooks found; a cookbook must at least have a metadata.rb file"
4. Look for errors in each cookbook in the list of cookbooks.

This should preserve existing behavior for people who use `chef-repo/cookbooks` and make it more usable for people who do the one-cookbook-per-repository model (e.g. the Berkshelf way).

I tried looking at the source, but I can't figure out how/why it is recursing into directories it doesn't know about.  If foodcritic is trying to detect cookbooks by looking for `recipes/` or `libraries/` then it should stop doing that and depend _only_ on the `metadata.rb` or `metadata.json` being at the root.

**Comment by SAPDanJoe** [comment]
I completely agree.  This would greatly increase the usability. Especially #3!  Have no output returned to the command line in both casses 1) when ther is no error AND 2)when there is no cookbook found; is very diffucult to understand.  Especially for new users.  With all the 'little things' that must be configured in Ruby, gems, chef, etc.etc.  getting no response at all leaves on wondering if you have suddenly become super-awesome and made no mistakes or if foodcritic is just another under-documented piece of the puzzle that you get to spend a few hours troubleshooting.
